src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -63,15 +63,23 @@ struct TreeTraversal<BVH, Predicates, Callback, SpatialPredicateTag>
     else
     {
       Kokkos::parallel_for("ArborX::TreeTraversal::spatial",
-                           Kokkos::RangePolicy<ExecutionSpace>(
+                           Kokkos::RangePolicy<ExecutionSpace, OrdinaryTree>(
                                space, 0, Access::size(predicates)),
                            *this);
     }
   }
 
+  KOKKOS_FUNCTION TreeTraversal(BVH const &bvh, Callback const &callback)
+      : _bvh{bvh}
+      , _callback{callback}
+  {}
+
   struct OneLeafTree
   {};
 
+  struct OrdinaryTree
```

Comment:
Perhaps `FullTree` instead of `OrdinaryTree`? Or `NontrivialTree`?

Reply:
FWIW I don't like FullTree at all and think "ordinary" (or "regular") was better